In [24]:
#Import Libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

In [25]:
#Input the Dataset
Ages = [14, 12, 11, 19 , 8 , 6 , 8] 
Ages_Birthday = [14, 12, 11, 19 , 8 , 7 , 8] 
Names = ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby', 'Cindy', 'Oliver']
Names_wJessica = ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby', 'Cindy', 'Jessica']
Ages_wJessica = [14, 12, 11, 19 , 8 , 7 , 1] 
index = [0, 1, 2, 3, 4, 5, 6]

In [26]:
#Make DataFrame
df = pd.DataFrame(index = index)
df['Names'] = Names
df['Ages'] = Ages
df['Ages w/ Birthday'] = Ages_Birthday
df['Names w /Jessica'] = Names_wJessica
df['Ages w/ Jessica'] = Ages_wJessica
df.head()

Names  Ages  Ages w/ Birthday Names w /Jessica  Ages w/ Jessica
0    Greg    14                14             Greg               14
1  Marcia    12                12           Marcia               12
2   Peter    11                11            Peter               11
3     Jan    19                19              Jan               19
4   Bobby     8                 8            Bobby                8

In [29]:
#Statistics on the DataFrame
Count = df.count()
Count = Count[0]
Mean_age, Median_age, Mode_age = df.mean(), df.median(), df.mode()
Variance_age = df.var()
StD_age = df.std()
StdError_age = StD_age / (Count ** 0.5)

7 Ages                11.142857
Ages w/ Birthday    11.285714
Ages w/ Jessica     10.285714
dtype: float64 Ages                11.0
Ages w/ Birthday    11.0
Ages w/ Jessica     11.0
dtype: float64 Ages                19.476190
Ages w/ Birthday    17.904762
Ages w/ Jessica     32.571429
dtype: float64 Ages                4.413184
Ages w/ Birthday    4.231402
Ages w/ Jessica     5.707138
dtype: float64 Ages                1.668027
Ages w/ Birthday    1.599320
Ages w/ Jessica     2.157096
dtype: float64
    Names  Ages  Ages w/ Birthday Names w /Jessica  Ages w/ Jessica
0   Bobby   8.0               8.0            Bobby                1
1   Cindy   NaN               NaN            Cindy                7
2    Greg   NaN               NaN             Greg                8
3     Jan   NaN               NaN              Jan               11
4  Marcia   NaN               NaN          Jessica               12
5  Oliver   NaN               NaN           Marcia               14
6   Peter   NaN   

# Using these estimates, if you had to choose only one estimate of central tendency and one estimate of variance to describe the data, which would you pick and why?

I would probably pick the Median and Standard Deviation. The median in this case is a good measure of the center and would not change significantly with the addition of an outlier. The standard deviation shows a possible distance away from the central data point an expected new data point may lie. 

# Next, Cindy has a birthday. Update your estimates- what changed, and what didn't?

Mean, St.Dev, St.Error, Variance are changed by the birthday. 
Mode, Median stay the same despite this change. This shows these variables are robust to small changes. 

# Nobody likes Cousin Oliver. Maybe the network should have used an even younger actor. Replace Cousin Oliver with 1-year-old Jessica, then recalculate again. Does this change your choice of central tendency or variance estimation methods?

No, my choice was designed to be robust against this type of change. 


# On the 50th anniversary of The Brady Bunch, four different magazines asked their readers whether they were fans of the show. The answers were: TV Guide 20% fans Entertainment Weekly 23% fans Pop Culture Today 17% fans SciPhi Phanatic 5% fans.

# Based on these numbers, what percentage of adult Americans would you estimate were Brady Bunch fans on the 50th anniversary of the show?

In [22]:
my_array = [20, 23, 17, 5]
my_DF = pd.DataFrame(my_array)
print(my_DF.median())

0    18.5
dtype: float64


I would estimate that ~18.5% are fans of the show. SciPhi Phanatic magazine's audience is an outlier and does not constitute the whole population of America. And so, the median is able to filter out most of the effects of the outlier in the dataset.   